python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -925,6 +924,36 @@ def get_event_records(
             except seven.JSONDecodeError:
                 logging.warning("Could not parse event record id `%s`.", row_id)
 
+        # If we haven't run the migration to add the event tags table, then it's possible that
+        # we haven't hit our limit. In that case, look further back.
+        if (
+            limit is not None
+            and num_results == limit
+            and num_unmigrated_tag_skips > 0
+            and event_records_filter.tags
+            and not self.has_table(AssetEventTagsTable.name)
+        ):
+            if ascending:
+                new_before_cursor = event_records_filter.before_cursor
+                new_after_cursor = row_id
+            else:
+                new_before_cursor = row_id
+                new_after_cursor = event_records_filter.after_cursor
+
+            event_records.extend(
```

Comment:
A little wary of exploding the stack by invoking this recursive call.

Also, if I pass in a limit of 100, and i get 99 back, then i'm resetting the call with a limit 1, which means i'll be recursing with a window size of 1.

Reply:
> Also, if I pass in a limit of 100, and i get 99 back, then i'm resetting the call with a limit 1, which means i'll be recursing with a window size of 1.

Do you mind spelling out the issue a little more? What's wrong with recursing with a window size of 1 in that case?

Reply:
Oh I see, like then we'll have a stack frame for every 1 record